api/src/main/java/ca/bc/gov/educ/api/gradstudent/service/GraduationStatusService.java

```diff
@@ -1349,18 +1364,18 @@ private GraduationStudentRecord processReceivedStudent(GraduationStudentRecord e
     }
 
     public List<GraduationStudentRecord> getStudentsForSchoolReport(String schoolOfRecord) {
-        return graduationStatusTransformer.tToDForBatch(graduationStatusRepository.findBySchoolOfRecord(schoolOfRecord));
+        return graduationStatusTransformer.tToDForBatchView(graduationStatusRepository.findBySchoolOfRecordAndStudentStatus(schoolOfRecord, "CUR"));
     }
 
     public List<UUID> getStudentsForAmalgamatedSchoolReport(String schoolOfRecord,String type) {
-        return graduationStatusTransformer.tToDForAmalgamation(graduationStatusRepository.findBySchoolOfRecordAmalgamated(schoolOfRecord),type);
+        return graduationStatusTransformer.tToDForAmalgamation(graduationStatusRepository.findBySchoolOfRecordAndStudentStatusAndStudentGradeIn(schoolOfRecord, "CUR", List.of("AD", "12")),type);
     }
 
-    public List<GraduationStudentRecord> updateStudentFlagReadyForBatchJobByStudentIDs(String batchJobType, List<UUID> studentIDs) {
+    public void updateStudentFlagReadyForBatchJobByStudentIDs(String batchJobType, List<UUID> studentIDs) {
         logger.debug("updateStudentFlagReadyForBatchJobByStudentIDs");
-        return studentIDs.stream()
-                .map(stid -> updateStudentFlagReadyForBatchJob(stid, batchJobType))
-                .filter(Objects::nonNull).collect(Collectors.toList());
+        for(UUID uuid: studentIDs) {
+            updateStudentFlagReadyForBatchJob(uuid, batchJobType);
```

Comment:
Please make updateStudentFlagReadyForBatchJob, and saveBatchFlagsOfGraduationStudentRecord void and remove graduationStatusTransformer.transformToDTOWithModifiedProgramCompletionDate(entity) from saveBatchFlagsOfGraduationStudentRecord. No need to transform to DTO when data not used in any calling methods.

Reply:
Fixed